backend-tests/tests/test_decommission.py

```diff
@@ -0,0 +1,275 @@
+                                           tenant_token)
+
+        r = devauthd.call('POST',
+                      deviceauth.URL_AUTH_REQS,
+                      body,
+                      headers=sighdr)
+        assert r.status_code == 200
+        dtoken = r.text
+
+        # wait for the device provisioning workflow to do its job
+        timeout = time.time() + 60
+        while time.time() < timeout:
+            r = invm.with_auth(utoken).call('GET',
+                                            inventory.URL_DEVICE,
+                                            path_params={'id': aset.did})
+            if r.status_code == 200:
+                break
+            else:
+                logger.debug("waiting for the device to be added to inventory...")
+                time.sleep(1)
+        else:
+            assert False, "device not added to the inventory"
+
+        # check if the device can access API by patching device inventory
+        payload = [
+                {
+                    "name": "mac",
+                    "value": "foo"
+                }
+        ]
+        r = invd.with_auth(dtoken).call('PATCH',
+                                    inventory.URL_DEVICE_ATTRIBUTES,
+                                    payload)
+        assert r.status_code == 200
+
+        # get all decommission_device WFs for reference
+        query = conductor.workflow_by_name_query(conductor.DECOMMISSIONING_DEVICE_WORKFLOW_NAME)
+        initial_wfs = conductor.get_workflows(params=query)
+        logger.debug("INITIAL WORKFLOWS: ", initial_wfs)
+
+        # decommission
+        r = devauthm.with_auth(utoken).call('DELETE',
+                          deviceauth_v2.URL_DEVICE.format(id=aset.did))
+
+        # check device is rejected
+        r = dc.with_auth(dtoken).call('GET',
+                                      deployments.URL_NEXT,
+                                      qs_params={'device_type': 'foo',
+                                                 'artifact_name': 'bar'})
+        assert r.status_code == 401
+
+        # check that the workflow completed successfully
+        timeout = time.time() + (60 * 5)
+        while time.time() < timeout:
+            wfs = conductor.get_workflows(params=query)
+            logger.debug("WORKFLOWS: ", wfs)
+            if wfs['totalHits'] == initial_wfs['totalHits'] + 1:
+                workflow = wfs['results'][0]
+                m = re.search('(?<=device_id=)\w+', workflow['input'])
+                devId = m.group(0)
+                if workflow['status'] == "COMPLETED" and devId == aset.did:
+                    logger.debug("decommissioning workflow completed")
+                    break
+                else:
+                    logger.debug("waiting for decommission_device workflow...")
+                    time.sleep(1)
+            else:
+                logger.debug("waiting for decommission_device workflow...")
+                time.sleep(1)
```

Comment:
As we are using the break on success, I think this can be simplified removing the else(s) and just having
```
if wfs['totalHits'] == initial_wfs['totalHits'] + 1:
    ...
    if ...
        break
logger.debug("waiting for decommission_device workflow...")
time.sleep(1)
```

Reply:
fixed